### ISSUE TITLE
[BUG] System.MissingMethodException No constructor found for CommunityToolkit.Maui.Behaviors.TouchBehavior+AccessibilityListener

### ISSUE THREAD

**Issue by semdelion** [CommunityToolkit/Maui#2950]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Did you read the "Reporting a bug" section on Contributing file?

- [x] I have read the "Reporting a bug" section on Contributing file: https://github.com/CommunityToolkit/Maui/blob/main/CONTRIBUTING.md#reporting-a-bug

### Current Behavior

This happens sometimes in production when the app was sent to background or resumes from background. I suppose that the system tries to restore an instance of AccessibilityListener using the specified constructor. This may be fixable by adding this constructor while doing the appropriate base call.

### Expected Behavior

No crash

### Steps To Reproduce

Unfortunately, I was unable to reproduce the problem.

### Link to public reproduction project repository

null

### Environment

```markdown
- .NET MAUI CommunityToolkit: 12.3.0
- OS: Android 11-15
- .NET MAUI: 9.0.120
```

### Anything else?

I reopened the issue https://github.com/CommunityToolkit/Maui/issues/2383
I'm seeing this error in the App Center logs. I couldn't reproduce the crash myself.

I assume a similar fix should help: https://github.com/xamarin/XamarinCommunityToolkit/commit/33c36af2c018d7981eb0931bfcd160168c77a38f

```
App Center Crash:
[10:07:00.177 mem:41885640 FTL] Unhandled Exception Error logged in MainApplication.cs->UnhandledExceptionRaiser:274 

 Subject: null 

 System.NotSupportedException: Unable to activate instance of type CommunityToolkit.Maui.Behaviors.TouchBehavior+AccessibilityListener from native handle 0x7fd420e438 (key_handle 0x4a71429).
 ---> System.MissingMethodException: No constructor found for CommunityToolkit.Maui.Behaviors.TouchBehavior+AccessibilityListener::.ctor(System.IntPtr, Android.Runtime.JniHandleOwnership)
 ---> Java.Interop.JavaLocationException: Exception_WasThrown, Java.Interop.JavaLocationException
Java.Lang.Error: Exception_WasThrown, Java.Lang.Error

  --- End of managed Java.Lang.Error stack trace ---
java.lang.Error: Java callstack:
	at crc640fd0ddb16fe433d4.TouchBehavior_AccessibilityListener.n_onAccessibilityStateChanged(Native Method)
	at crc640fd0ddb16fe433d4.TouchBehavior_AccessibilityListener.onAccessibilityStateChanged(TouchBehavior_AccessibilityListener.java:31)
	at android.view.accessibility.AccessibilityManager.lambda$notifyAccessibilityStateChanged$0(AccessibilityManager.java:1898)
	at android.view.accessibility.AccessibilityManager$$ExternalSyntheticLambda1.run(Unknown Source:4)
	at android.os.Handler.handleCallback(Handler.java:942)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loopOnce(Looper.java:240)
	at android.os.Looper.loop(Looper.java:351)
	at android.app.ActivityThread.main(ActivityThread.java:8423)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:584)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1013)
```
